cswinrt/strings/additions/Windows.Storage.Streams/WindowsRuntimeBufferExtensions.cs

```diff
@@ -118,6 +119,7 @@ public static byte[] ToArray(this IBuffer source)
         public static byte[] ToArray(this IBuffer source, uint sourceIndex, int count)
         {
             if (source == null) throw new ArgumentNullException(nameof(source));
+            if (source.Length == 0) return Array.Empty<byte>();
```

Comment:
I feel like this scenario should technically error.  We are told to convert to an array with the array starting with data from source index to a length of count.   In the length == 0 case and count != 0, we return a empty array which might unexpected to the caller and maybe even against what the API was documented as.

We should probably do source.length == 0 and count == 0, return empty.  otherwise throw exception like before.